### PR TITLE
DNN-9408: Upgrade ClientDependency.Core source code to latest

### PR DESCRIPTION
**Pull request by galatrash-at-dnn**

### CODE REVIEW

File: DNN Platform/Components/ClientDependency/Source/Controls/ClientDependencyInclude.cs [hunk 1]
```diff
@@ -58,31 +60,27 @@ protected ClientDependencyInclude(IClientDependencyFile file)
...
-            {
                 throw new NullReferenceException("Both File and Type properties must be set");
-            }
-        }
-
-        protected override void Render(HtmlTextWriter writer)
-        {
-            if (AddTag || this.Context.IsDebuggingEnabled)
-            {
-                writer.Write("<!--CDF({0}|{1})-->", DependencyType, FilePath);
```
Comment (bdukes) [inline DNN Platform/Components/ClientDependency/Source/Controls/ClientDependencyInclude.cs]: We'll want to add this piece back, it's an addition that DNN uses when a module is cached

File: DNN Platform/Components/ClientDependency/Source/DependencySorter.cs [hunk 2]
```diff
@@ -10,30 +10,6 @@ namespace ClientDependency.Core
     /// </summary>
     internal static class DependencySorter
     {
-
-        public static IList<IClientDependencyFile> FilterDependencies(IList<IClientDependencyFile> dependencies)
```
Comment (bdukes) [inline DNN Platform/Components/ClientDependency/Source/DependencySorter.cs]: I believe this is an addition from @donker which allows framework deduplication

File: DNN Platform/Components/ClientDependency/Source/FileRegistration/Providers/BaseFileRegistrationProvider.cs [hunk 3]
```diff
@@ -461,11 +490,11 @@ private string AppendVersion(string url, HttpContextBase http)
             {
                 //sort both the js and css dependencies properly
 
-                var jsDependencies = DependencySorter.SortItems(DependencySorter.FilterDependencies(
```
Comment (bdukes) [inline DNN Platform/Components/ClientDependency/Source/FileRegistration/Providers/BaseFileRegistrationProvider.cs]: This uses the piece above for framework deduplication

File: DNN Platform/Components/ClientDependency/Source/Controls/ClientDependencyInclude.cs (hunk 1 above)
Comment (galatrash-at-dnn) [reply DNN Platform/Components/ClientDependency/Source/Controls/ClientDependencyInclude.cs]: Code restored.

File: DNN Platform/Components/ClientDependency/Source/DependencySorter.cs (hunk 2 above)
Comment (galatrash-at-dnn) [reply DNN Platform/Components/ClientDependency/Source/DependencySorter.cs]: Code restored.

File: DNN Platform/Components/ClientDependency/Source/FileRegistration/Providers/BaseFileRegistrationProvider.cs (hunk 3 above)
Comment (galatrash-at-dnn) [reply DNN Platform/Components/ClientDependency/Source/FileRegistration/Providers/BaseFileRegistrationProvider.cs]: Code restored.

**Comment by galatrash-at-dnn** [comment]
I restored all pieces of code referenced by @bdukes and it is ready for another review.

**Comment by zyhfish** [comment]
Changes looks good and merged.

**Pull request merged**